client/my-sites/post-type-list/post-actions-ellipsis-menu/qrcode.jsx

```diff
@@ -0,0 +1,65 @@
+					fill-color="#000000"
+				/>
+			</svg>
+		}
+	/>
+);
+
+function PostActionsEllipsisMenuQRCode( { translate, status, url } ) {
+	const [ showQRCode, setShowQRCode ] = useState( false );
+
+	const generateQRCode = () => {
+		setShowQRCode( true );
+	};
+
+	if ( ! url || ! status ) {
+		return null;
+	}
+
+	// The QR option will be available only for `publish` status
+	if ( status !== 'publish' ) {
+		return null;
+	}
+
+	return (
+		<>
+			<PopoverMenuItem onClick={ generateQRCode } icon={ qrIcon }>
+				{ translate( 'QR Code' ) }
+			</PopoverMenuItem>
+			<PostActionsQRCode siteUrl={ url } showQRCode={ showQRCode } />
+		</>
+	);
+}
+
+PostActionsEllipsisMenuQRCode.propTypes = {
+	globalId: PropTypes.string.isRequired,
+};
+
+const mapStateToProps = ( state, { globalId } ) => {
+	const post = getPost( state, globalId );
+	if ( ! post ) {
+		return {};
+	}
+
+	return {
+		status: post.status,
+		url: post.URL,
+	};
+};
+
+export default connect( mapStateToProps )( localize( PostActionsEllipsisMenuQRCode ) );
```

Comment:
Just as with `connect()`, the `localize()` HoC is not recommended for use in functional components. Instead, we use the `useTranslate()` hook directly in the component.

Reply:
Now we are using `useTranslate()`. Thanks for the recommendation!